node/avn-service/src/ethereum_events_handler.rs

```diff
@@ -721,6 +770,17 @@ where
         )
         .map_err(|err| format!("Failed to check if author has casted event vote: {:?}", err))?;
 
+    let additional_events: Vec<AdditionalEvent> = config
+        .client
+        .runtime_api()
+        .additional_events(config.client.info().best_hash)
+        .map_err(|err| format!("Failed to query event signatures: {:?}", err))?
+        .iter()
+        .map(|events_set| events_set.iter().collect::<Vec<_>>())
```

Comment:
`fn additional_events()` returns an `Option<AdditionalEvents>`. AdditionalEvents is a `BoundedBTreeSet<AdditionalEvent, ConstU32<16>>` and `AdditionalEvent` is a struct. So what is an `events_set`?

Reply:
If we have an inner array, maybe using `.flat_map(|events_set| events_set.iter())` instead of `map` and `flatten` would make it more efficient and easier to understand.

Reply:
you can iterate an option.
event_set is an `AdditionalEvents` object which is a BoundedBTreeSet.